File: src/main/java/dev/arbor/gtnn/data/GTNNDataGen.java
package dev.arbor.gtnn.data;

import com.tterrag.registrate.providers.ProviderType;
import dev.arbor.gtnn.api.registry.CNLangProvider;
import dev.arbor.gtnn.data.lang.LangHandler;

import static dev.arbor.gtnn.GTNNRegistries.REGISTRATE;

public class GTNNDataGen {
    public static final ProviderType<CNLangProvider> CN_LANG = ProviderType.register("cn_lang", (p, e) -> new CNLangProvider(p, e.getGenerator().getPackOutput()));
    public static void init() {
        REGISTRATE.addDataGenerator(ProviderType.LANG, LangHandler::enInitialize);
        REGISTRATE.addDataGenerator(CN_LANG, LangHandler::cnInitialize);
        REGISTRATE.addDataGenerator(ProviderType.BLOCK_TAGS, GTNNTags::initBlock);
    }

}

File: src/main/java/dev/arbor/gtnn/data/GTNNDataGen.kt
package dev.arbor.gtnn.data

import com.tterrag.registrate.AbstractRegistrate
import com.tterrag.registrate.providers.ProviderType
import com.tterrag.registrate.providers.RegistrateLangProvider
import com.tterrag.registrate.providers.RegistrateTagsProvider.IntrinsicImpl
import com.tterrag.registrate.util.nullness.NonNullConsumer
import dev.arbor.gtnn.api.registry.CNLangProvider
import dev.arbor.gtnn.data.lang.LangHandler
import net.minecraft.world.level.block.Block
import net.minecraftforge.data.event.GatherDataEvent
import dev.arbor.gtnn.GTNNRegistries.REGISTRATE

object GTNNDataGen {
    val CN_LANG: ProviderType<CNLangProvider> = ProviderType.register(
        "cn_lang"
    ) { p: AbstractRegistrate<*>?, e: GatherDataEvent ->
        CNLangProvider(
            p!!, e.generator.packOutput
        )
    }

    fun init() {
        REGISTRATE.addDataGenerator(ProviderType.LANG
        ) { provider: RegistrateLangProvider ->
            LangHandler.enInitialize(
                provider
            )
        }
        REGISTRATE.addDataGenerator(CN_LANG
        ) { provider: CNLangProvider ->
            LangHandler.cnInitialize(
                provider
            )
        }
        REGISTRATE.addDataGenerator<IntrinsicImpl<Block>>(ProviderType.BLOCK_TAGS
        ) { provider: IntrinsicImpl<Block> ->
            GTNNTags.initBlock(
                provider
            )
        }
    }

}
